src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -245,7 +245,8 @@ var ReactCompositeComponentMixin = {
     inst.props = publicProps;
     inst.context = publicContext;
     inst.refs = emptyObject;
-    inst.updater = ReactUpdateQueue;
+    inst.updater = transaction.getUpdateQueue ?
```

Comment:
This should be a required method—there is no need to check for its existence. It’s bad when similar objects have different shapes because it makes JIT optimizations harder.